src/main/scala/org/dbpedia/spotlight/util/WikiUtil.scala

```diff
@@ -0,0 +1,121 @@
+ *  limitations under the License.
+ */
+
+package org.dbpedia.spotlight.util
+
+//import org.dbpedia.extraction.util.RichString.wrapString
+//import org.dbpedia.util.text.uri.UriDecoder
+import java.util.Arrays
+
+/**
+ * Contains several utility functions related to WikiText.
+ */
+object WikiUtil
+{
+  /**
+   * replace underscores by spaces, replace non-breaking space by normal space, remove
+   * exotic whitespace, normalize duplicate spaces, trim whitespace (any char <= U+0020)
+   * from start and end.
+   *
+   * Also see WikiTitle.parse().
+   *
+   * TODO: better treatment of U+20xx: remove some, replace some by space, others by LF
+   *
+   * FIXME: There is no logic to our decoding / encoding of strings, URIs, etc. It's done
+   * in too many places. We must set a policy and use distinct classes, not generic strings.
+   *
+   * string string possibly using '_' instead of ' '
+   */
+  /*def cleanSpace(string: String): String =
+  {
+    // FIXME: removing these chars may avoid some errors, but also introduces others.
+    // For example, the local name of the 'Project' namespace on fa wikipedia contains U+200C.
+    string.replaceChars("_\u00A0\u200E\u200F\u2028\u202A\u202B\u202C\u3000", "  ").replaceAll(" +", " ").trim
+  }*/
+
+  /**
+   * Replacement string array for StringUtils.escape
+   */
+  //def iriReplacements: Array[String] = StringUtils.replacements('%', "\"#%<>?[\\]^`{|}")
+
+  // store our own private copy of the mutable array
+  //private val replacements = iriReplacements
+
+  /**
+   * Replaces multiple spaces (U+0020) by one, removes spaces from start and end,
+   * replaces spaces by underscores, and percent-encodes the following characters:
+   *
+   *
+   * The result is usable in most parts of a IRI. The ampersand '&' is not escaped though.
+   *
+   * Should only be used for canonical MediaWiki page names. Not for fragments, not for queries.
+   *
+   * TODO: a canonical MediaWiki page name does not contain multiple spaces. We should not
+   * clean spaces but simply throw an exception if the name is not canonical.
+   *
+   * @param name Canonical MediaWiki page name, e.g. 'Émile Zola'
+   */
+  def wikiEncode(name : String) : String=
+  {
+    // TODO: all this replacing is inefficient, one loop over the string would be nicer.
+
+    // replace spaces by underscores.
+    // Note: MediaWiki apparently replaces only spaces by underscores, not other whitespace.
+    var encoded = name.replace(' ', '_');
+
+    // normalize duplicate underscores
+    encoded = encoded.replaceAll("_+", "_");
+
+    // trim underscores from start
+    encoded = encoded.replaceAll("^_", "");
+
+    // trim underscores from end
+    encoded = encoded.replaceAll("_$", "");
+
+    return encoded
+    //StringUtils.escape(encoded, replacements)
+  }
+
+
+  /**
+   * name encoded MediaWiki page name, e.g. '%C3%89mile_Zola'.
+   * Must not include the namespace (e.g. 'Template:').
+   */
+  /*def wikiDecode(name : String) : String =
```

Comment:
Yes. This is not needed. I was able to call wikiutils from dbpedia-extraction repo.
I will remove this.